### PR TITLE
feat(styling and theme): Added Parameter for Styling and Theme

### PR DESCRIPTION
**Pull request by gaokevin1**
## Description

This PR adds the ability to control styling and theme from a URL query parameter in the Auth Hosting application.

## Must

- [ ] 📱 Responsiveness (mobile/XL resolutions)
- [ ] 🧪 Tests
- [ ] 📃 Documentation (if applicable)

### CODE REVIEW

**Review by Bars92 — APPROVED** [review]

File: src/App.tsx
Comment (asafshen) [inline src/App.tsx:8]: This is already defined on the sdk , we can take it from there

File: src/App.tsx
Comment (asafshen) [inline src/App.tsx:78]: Light is not a good default, let's leave it to the sdk to handle the default

File: src/App.tsx
Comment (asafshen) [inline src/App.tsx:81]: FYI , We don't need to cast each var, we can cast them all together, it will be more readable and less repetitive

File: src/App.tsx
Comment (orius123) [inline src/App.tsx:78]: We already have this code in the dynamicSdk branch
Do we really need it also here?

File: src/App.tsx
Comment (orius123) [inline src/App.tsx:109]: isn't it `style-id` ?

File: src/App.tsx
Comment (gaokevin1) [reply src/App.tsx:78]: I think it's important in case someone clones the repo, and expects to follow our docs to be able to use this functionality.

**Comment by gaokevin1** [comment]
Btw @Bars92 `style-id` doesn't seem to work... we definitely should not merge this until it works.

Also @asafshen, do you want another test for this PR besides this one?

```
test('that the flow can be customized with search params', async () => {
		window.location.pathname = `/${packageJson.homepage}/${validProjectId}`;
		window.location.search = `?debug=${debug}&flow=${flowId}`;
		render(<App />);
		expect(mockDescope).toHaveBeenCalledWith(
			expect.objectContaining({ debug, flowId })
		);
	});
```

**Comment by gaokevin1** [comment]
I'm also of the opinion, along with @gshriki , that the flow parameter should be `styleId` instead of `style-id`.

**Comment by asafshen** [comment]
from what I see in the PR in descope-js this was not added yet to react-sdk https://github.com/descope/descope-js/pull/744/files#diff-b24066d0d30844fccfe74b02e0ee2dedf42a1cf7fa1bd6377fe6fd08695fd234
only to web-component at this stage

**Comment by gaokevin1** [comment]
Yeah we'll hold off until it's added to the React SDK.

File: src/App.tsx
Comment (orius123) [inline src/App.tsx:106]: This is calling the react-sdk so it will be `styleId`
https://github.com/descope/descope-js/pull/782/files

File: src/App.tsx
Comment (orius123) [reply src/App.tsx]: Bot sure I agree, but anyhow the code. should be the same or we will never be able to merge back once we get rid of dynamic SDK

**Comment by gaokevin1** [comment]
@orius123 can you please review this and merge now?

**Comment by orius123** [comment]
@gaokevin1 build is failing

**Comment by gaokevin1** [comment]
@orius123 can you please review again? Build should be passing now.

**Review by orius123 — DISMISSED** [review]
All deployments fail

**Comment by gaokevin1** [comment]
That's because of my vercel access not because of the build. If you push something you'll see.

**Review by asafshen — APPROVED** [review]

**Comment by dorsha** [comment]
This was not added to the dynamicSdk, which means it is not available to customers.
So we have a doc that was not tested in production.
image

**Comment by gshriki** [comment]
interesting. I was sure @anvibanga checked it...

**Comment by anvibanga** [comment]
@chris4490 @gaokevin1

**Comment by chris4490** [comment]
Yea I thought I'd checked it as well.

**Pull request merged**
